src/meepgeom.cpp

```diff
@@ -1765,17 +1768,30 @@ void fragment_stats::compute_stats() {
     geometric_object *go = &geom.items[i];
     double overlap = box_overlap_with_object(box, *go, tol, maxeval);
 
+    // If the object doesn't overlap the entire box, that implies that
+    // an object interface intercepts the box, which means we treat
+    // the entire box as anisotropic.
```

Comment:
Maybe add a comment that this could give some false positives if there is another object with the same material behind the current object, but in practice it is probably reasonable to assume that there is a material interface somewhere in the box so we won't worry about fancier edge-detection methods for now.